### ISSUE TITLE
Improve VacuousTest check

### ISSUE THREAD

**Issue by s3cur3** [Jump-App/credo_checks#1]
`Jump.CredoChecks.VacuousTest` can generate both false positives and false negatives. Examples:

## False negatives

Our alias tracking doesn't support `alias ..., as: ...`, so you might do something like `alias Phoenix, as: Phx`, causing us to think `Phx` is a module in your own codebase.

## False positives

Tests that genuinely exercise Protocol or Behaviour implementations currently get flagged as vacuous. Examples:

```elixir
test "does not include password" do
  refute inspect(%User{password: "123456"}) =~ "password: \"123456\""
end
```

```elixir
test "connection fails with an invalid token" do
  assert Phoenix.ChannelTest.connect(UserSocket, %{"token" => "invalid"}) == :error
end
```

I'm not sure what to do with these, because within the static analysis context, I don't think we can reliably identify that the function being called depends on such-and-such Behaviour/Protocol, which the specified module implements.

One option would be something like a "relaxed" mode to the check configuration which then says "as long as you use an app-defined module at all, the test is good." I worry that will make the check basically useless, though... it would let things like this through:

```elixir
test "user has the desired fields", %{user: user} do
  assert %User{id: _} = user
end
```

Another option would be to support a tag (`@tag`, `@describetag`, `@moduletag`) that says "this is a Behaviour test, so chill out." If it requires manually tagging tests, though, at that point you might as well just add:

```elixir
# credo:disable-for-next-line Jump.CredoChecks.VacuousTest
```

...or:

```elixir
# credo:disable-for-this-file Jump.CredoChecks.VacuousTest
```

I'm certainly open to ideas!
